Source/Core/DolphinQt/Debugger/NetworkWidget.cpp

```diff
@@ -0,0 +1,299 @@
+  QStringList header{tr("ID"), tr("Domain"), tr("Type"), tr("State"), tr("Name")};
+  m_ssl_table->setColumnCount(header.size());
+
+  m_ssl_table->setHorizontalHeaderLabels(header);
+  m_ssl_table->setTabKeyNavigation(false);
+  m_ssl_table->verticalHeader()->setVisible(false);
+  m_ssl_table->setEditTriggers(QAbstractItemView::NoEditTriggers);
+  m_ssl_table->setSelectionMode(QAbstractItemView::NoSelection);
+  m_ssl_table->setWordWrap(false);
+
+  ssl_context_layout->addWidget(m_ssl_table, 0, 0);
+  ssl_context_layout->setSpacing(1);
+  return ssl_context_group;
+}
+
+QGroupBox* NetworkWidget::CreateSSLOptionsGroup()
+{
+  QGroupBox* ssl_options_group = new QGroupBox(tr("SSL options"));
+  QGridLayout* ssl_options_layout = new QGridLayout;
+  ssl_options_group->setLayout(ssl_options_layout);
+
+  m_dump_ssl_read_checkbox = new QCheckBox(tr("Dump SSL read"));
+  m_dump_ssl_write_checkbox = new QCheckBox(tr("Dump SSL write"));
+  m_dump_root_ca_checkbox = new QCheckBox(tr("Dump root CA"));
+  m_dump_peer_cert_checkbox = new QCheckBox(tr("Dump peer certificates"));
+  m_verify_certificates_checkbox = new QCheckBox(tr("Verify certificates"));
+
+  ssl_options_layout->addWidget(m_dump_ssl_read_checkbox, 0, 0);
+  ssl_options_layout->addWidget(m_dump_ssl_write_checkbox, 1, 0);
+  ssl_options_layout->addWidget(m_verify_certificates_checkbox, 2, 0);
+  ssl_options_layout->addWidget(m_dump_root_ca_checkbox, 0, 1);
+  ssl_options_layout->addWidget(m_dump_peer_cert_checkbox, 1, 1);
+
+  ssl_options_layout->setSpacing(1);
+  return ssl_options_group;
+}
+
+QTableWidgetItem* NetworkWidget::GetSocketDomain(s32 host_fd) const
+{
+  if (host_fd < 0)
+    return new QTableWidgetItem();
+
+  sockaddr sa;
+  socklen_t sa_len = sizeof(sa);
+  const int ret = getsockname(host_fd, &sa, &sa_len);
+  if (ret != 0)
+    return new QTableWidgetItem(tr("Unknown"));
+
+  switch (sa.sa_family)
+  {
+  case 2:
+    return new QTableWidgetItem(QLatin1Literal("AF_INET"));
+  case 23:
+    return new QTableWidgetItem(QLatin1Literal("AF_INET6"));
+  default:
+    return new QTableWidgetItem(QString::number(sa.sa_family));
+  }
+}
+
+QTableWidgetItem* NetworkWidget::GetSocketType(s32 host_fd) const
+{
+  if (host_fd < 0)
+    return new QTableWidgetItem();
+
+  int so_type;
+  socklen_t opt_len = sizeof(so_type);
+  const int ret =
+      getsockopt(host_fd, SOL_SOCKET, SO_TYPE, reinterpret_cast<char*>(&so_type), &opt_len);
+  if (ret != 0)
+    return new QTableWidgetItem(tr("Unknown"));
+
+  switch (so_type)
+  {
+  case 1:
+    return new QTableWidgetItem(QLatin1Literal("SOCK_STREAM"));
+  case 2:
+    return new QTableWidgetItem(QLatin1Literal("SOCK_DGRAM"));
+  default:
+    return new QTableWidgetItem(QString::number(so_type));
+  }
+}
+
+QTableWidgetItem* NetworkWidget::GetSocketState(s32 host_fd) const
+{
+  if (host_fd < 0)
+    return new QTableWidgetItem();
+
+  struct sockaddr_in peer_addr;
+  socklen_t peer_addr_len = sizeof(struct sockaddr_in);
```

Comment:
```suggestion
  socklen_t peer_addr_len = sizeof(sockaddr_in);
```

or even `sizeof(peer_addr)`